Modules/ZDC/src/ZDCRawDataTask.cxx

```diff
@@ -0,0 +1,1032 @@
+                         //fMapBinNameIdSummaryHisto.insert(std::pair<std::string,int>(chName,i));
+                         //ILOG(Info, Support) << TString::Format("{%d}[%d][%d] %s ",i,imod,ich,chName.c_str()) << ENDM; // QcInfoLogger is used. FairMQ logs will go to there as well.
+                     }
+                }
+            }
+            getObjectsManager()->startPublishing(fSummaryPedestal);
+            try {
+              getObjectsManager()->addMetadata(fSummaryPedestal->GetName(), "custom", "34");
+              return true;
+            }
+            catch (...) {
+              ILOG(Warning, Support) << "Metadata could not be added to " << fSummaryPedestal->GetName() << ENDM;
+              return false;
+            }
+        }
+
+    }
+    return false;
+}
+
+
+
+// Decode Configuration file
+
+std::vector<std::string> ZDCRawDataTask::tokenLine(std::string Line, std::string Delimiter){
+    std::string token;
+    size_t pos = 0;
+    int i=0;
+    std::vector<std::string> stringToken;
+    while ((pos = Line.find(Delimiter)) != std::string::npos) {
+        token = Line.substr(i, pos);
+        stringToken.push_back(token);
+        Line.erase(0, pos + Delimiter.length());
+    }
+    stringToken.push_back(Line);
+    return stringToken;
+}
+
+std::string ZDCRawDataTask::removeSpaces( std::string s )
+  {
+  std::string result = boost::trim_copy( s );
+  while (result.find( " " ) != result.npos)
+    boost::replace_all( result, " ", "" );
+  return result;
+  }
+
+bool ZDCRawDataTask::configureRawDataTask() {
+
+    std::ifstream file(std::getenv("CONF_HISTO_QCZDCRAW"));
+    std::string line;
+    std::vector<std::string> tokenString;
+    int line_number=0;
+    bool check;
+    bool error = true;
+    if(!file){
+       initHisto();
+        }
+    else{
+        ILOG(Info, Support) << "initialize ZDC RAW DATA HISTOGRAMS FROM FILE" << ENDM;
+        while(getline(file, line)){
+            line_number ++;
+            tokenString = tokenLine(line,";");
+            ILOG(Info, Support) << TString::Format("Decode Line number %d %s",line_number, line.c_str()) << ENDM;
+            if (line.compare(0,1,"#") == 0 || line.compare(0,1,"\n") == 0 || line.compare(0,1," ") == 0 || line.compare(0,1,"\t") == 0 || line.compare(0,1,"") == 0  ) check = true;
+            else check=decodeConfLine(tokenString,line_number);
+            if(check == false){
+                error = false;
+            }
+            tokenString.clear();
+        }
+    }
+    if (error == false) initHisto();
+    file.close();
+    return true;
+}
+
+bool ZDCRawDataTask::decodeConfLine(std::vector<std::string> TokenString,int LineNumber){
+    // Module Mapping
+    ILOG(Info, Support) << TString::Format("Key Word %s",TokenString.at(0).c_str()) << ENDM;
+
+    if (strcmp(TokenString.at(0).c_str(), "MODULE") == 0) {
+        if (decodeModule(TokenString,LineNumber) == false) return false;
+        else return true;
```

Comment:
```suggestion
        return decodeModule(TokenString,LineNumber);
```
can't all of those be simplified to something like this?

Reply:
Yes I can change it